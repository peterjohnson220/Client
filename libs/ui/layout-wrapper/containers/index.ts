export * from './layout-wrapper';
export * from './left-sidebar';
