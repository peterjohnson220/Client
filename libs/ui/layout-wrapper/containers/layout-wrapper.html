<!--<div class="flex-column-container">
  <div class="header">
    <pf-layout-wrapper-header [avatarSource]="avatarSource"
                              [companyLogoSource]="companyLogoSource"
                              [dropdownNavigationLinks]="headerDropdownNaivgationLinks$ | async"
                              [userContext]="userContext$ | async">
    </pf-layout-wrapper-header>
  </div>
  <pf-layout-wrapper-left-sidebar></pf-layout-wrapper-left-sidebar>
  <pf-layout-wrapper-right-sidebar></pf-layout-wrapper-right-sidebar>
  <div class="content">
    <ng-content></ng-content>
  </div>
</div>-->

<div class="container-fluid">
  <pf-layout-wrapper-header [avatarSource]="avatarSource"
                            [companyLogoSource]="companyLogoSource"
                            [dropdownNavigationLinks]="headerDropdownNaivgationLinks$ | async"
                            [userContext]="userContext$ | async">
  </pf-layout-wrapper-header>
  <div id="wrapper" class="page-container" [class.toggledLeft]="leftSidebarToggle" style="padding-left: 57px;">
    <div id="left-sidebar-wrapper" [class.left-sidebar-enabled]="leftSidebarToggle">
      <div id="left-sidebar-dock">
      </div>
      <pf-layout-wrapper-left-sidebar [leftSidebarToggle]="leftSidebarToggle"></pf-layout-wrapper-left-sidebar>
      <div id="left-sidebar-footer" style="background-color: #283B4A; width:52px; position: fixed; left: 0; transition: width .5s;" [class.left-sidebar-enabled]="leftSidebarToggle">
        <i (click)="leftSidebarToggle = !leftSidebarToggle" id="left-sidebar-toggle" class="fa fa-lg" [class.left-sidebar-enabled-footer]="leftSidebarToggle" [ngClass]="{'fa-angle-double-left': leftSidebarToggle, 'fa-angle-double-right': !leftSidebarToggle}" style="position:absolute; left: 20px; bottom: 10px; transition: left .5s;"></i>
      </div>
    </div>
    <div id="page-content-wrapper" class="page-container">
      <ng-content></ng-content>
    </div>
    <div id="right-sidebar-wrapper">
      <pf-layout-wrapper-right-sidebar></pf-layout-wrapper-right-sidebar>
    </div>
  </div>
</div>
