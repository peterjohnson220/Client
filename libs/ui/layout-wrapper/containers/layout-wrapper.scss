.header{
  height: 47px;
  flex-shrink: 0;
}

.content{
  height: 100%;
  overflow-y: auto;
}
