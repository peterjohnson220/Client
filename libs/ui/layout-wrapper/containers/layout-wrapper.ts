import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'pf-layout-wrapper',
  templateUrl: './layout-wrapper.html',
  styleUrls: [ './layout-wrapper.scss' ]
})
export class LayoutWrapperComponent implements OnInit {
  constructor() {
  }

  ngOnInit() {

  }
}
