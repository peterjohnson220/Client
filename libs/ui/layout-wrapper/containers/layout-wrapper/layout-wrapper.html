<div class="wrapper">
  <header>
    <div class="flex-row-nowrap">
      <pf-layout-wrapper-header [dropdownNavigationLinks]="headerDropdownNavigationLinks$ | async"
                                [userContext]="userContext$ | async"
                                [homePageLink]="homePageLink$ | async"
                                [enableCoreJdmInClient]="enableCoreJdmInClient"
                                [requireSSOLogin]="requireSSOLogin$ | async">
      </pf-layout-wrapper-header>
    </div>
  </header>
  <main [class.main-body-scroll]="!centerContentScroll" [class.left-sidebar-expanded]="leftSidebarToggle">
    <div class="left-sidebar"
         [class.left-sidebar-center-content-scroll]="centerContentScroll"
         [class.d-none]="(userContext$ | async)?.IsPublic || (userContext$ | async)?.WorkflowStepInfo"
         (mouseenter)="leftSidebarToggleChanged(true)"
         (mouseleave)="leftSidebarToggleChanged(false)">
      <pf-layout-wrapper-left-sidebar [leftSidebarToggle]="leftSidebarToggle" [enableCoreJdmInClient]="enableCoreJdmInClient"></pf-layout-wrapper-left-sidebar>
    </div>
    <div class="center-content"
         [ngClass]="{ 'center-content-body-scroll': !centerContentScroll, 'center-content-scroll': centerContentScroll }">
      <ng-content select="[mainContent]"></ng-content>
    </div>
    <div class="right-sidebar" *ngIf="displayRightSideBar">
      <pf-layout-wrapper-right-sidebar
        [fontAwesomeOpenIcon]="rightSideBarFontAwesomeOpenIcon"
        [isOpen]="rightSidebarShouldBeOpen"
        (onToggle)="rightSidebarToggle($event)"
      >
        <ng-content select="[rightSidebarContent]"></ng-content>
      </pf-layout-wrapper-right-sidebar>
    </div>
  </main>
</div>
