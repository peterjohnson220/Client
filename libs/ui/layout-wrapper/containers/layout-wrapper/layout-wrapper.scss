@import 'colors';

.wrapper {
  display: flex;
  flex-flow: column nowrap;
  height: 100vh;
  min-height: 100vh;
}

.fixed-height{
  height: 100%;
}

header {
  height: 47px;
  flex: none;
}

main {
  flex: 1 0 auto;
}

.flex-row-nowrap {
  display: flex;
  flex-flow: row nowrap;
}

.flex-1 {
  flex: 1;
}

.left-sidebar {
  background: $left-sidebar-color;
}

.right-sidebar {

}
