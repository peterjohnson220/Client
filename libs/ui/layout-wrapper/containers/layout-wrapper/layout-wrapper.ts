import { Component, Input, OnInit } from '@angular/core';

import { Store } from '@ngrx/store';
import { Observable } from 'rxjs/Observable';

import * as fromRootState from '../../../../state/state';

import * as fromHeaderActions from '../../actions/header.actions';
import { UserContext, NavigationLink, HomePageLink } from '../../../../models';
import * as fromLayoutReducer from '../../reducers';

@Component({
  selector: 'pf-layout-wrapper',
  templateUrl: './layout-wrapper.html',
  styleUrls: ['./layout-wrapper.scss']
})
export class LayoutWrapperComponent implements OnInit {
  userContext$: Observable<UserContext>;
  displayRightSideBar$: Observable<boolean>;
  // Loading/Errors
  gettingHeaderDropdownNavigationLinks$: Observable<boolean>;
  gettingHeaderDropdownNavigationLinksError$: Observable<boolean>;
  headerDropdownNavigationLinks$: Observable<NavigationLink[]>;
  getGettingHomePageLink$: Observable<boolean>;
  getGettingHomePageLinkError$: Observable<boolean>;

  homePageLink$: Observable<HomePageLink>;

  @Input() displayRightSideBar: boolean;

  constructor(
    private store: Store<fromRootState.State>,
    private layoutStore: Store<fromLayoutReducer.LayoutWrapperState>

  ) {
    // Loading / Errors
    this.getGettingHomePageLink$ = this.layoutStore.select(fromLayoutReducer.getGettingHomePageLink);
    this.getGettingHomePageLinkError$ = this.layoutStore.select(fromLayoutReducer.getGettingHomePageLinkError);
    this.homePageLink$ = this.layoutStore.select(fromLayoutReducer.getHomePageLink);

    this.gettingHeaderDropdownNavigationLinks$ = layoutStore.select(fromLayoutReducer.getGettingHeaderDropdownNavigationLinks);
    this.gettingHeaderDropdownNavigationLinksError$ = layoutStore.select(fromLayoutReducer.getGettingHeaderDropdownNavigationLinksError);

    this.userContext$ = store.select(fromRootState.getUserContext);
    this.headerDropdownNavigationLinks$ = layoutStore.select(fromLayoutReducer.getHeaderDropdownNavigationLinks);
  }

  ngOnInit() {
    this.store.dispatch(new fromHeaderActions.GetHeaderDropdownNavigationLinks());
    this.userContext$.take(1).subscribe(i => {
      this.store.dispatch(new fromHeaderActions.GetHeaderUserHomePageLink({
        userId: i.UserId
      }));
    });
  }
}
