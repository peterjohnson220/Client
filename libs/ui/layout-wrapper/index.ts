export * from './components';
export * from './containers';
export * from './layout-wrapper.module';
