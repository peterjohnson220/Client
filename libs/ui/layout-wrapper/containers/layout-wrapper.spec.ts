import { Component, NO_ERRORS_SCHEMA } from '@angular/core';

import { TestBed, ComponentFixture } from '@angular/core/testing';
import { LayoutWrapperComponent } from './layout-wrapper';

// Host Component for testing transclusion
@Component({
  template: `
    <pf-layout-wrapper>
      <ng-container mainContent>I am the main content</ng-container>
      <ng-container rightSidebarContent>I am the right sidebar content</ng-container>
    </pf-layout-wrapper>\`
  `
})
class TestHostComponent {
}

describe('Layout Wrapper', () => {
  let fixture: ComponentFixture<LayoutWrapperComponent>;
  let instance: LayoutWrapperComponent;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        LayoutWrapperComponent, TestHostComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    fixture = TestBed.createComponent(LayoutWrapperComponent);
    instance = fixture.componentInstance;
  });

  it('should transclude', () => {
    const hostComponent = TestBed.createComponent(TestHostComponent);

    fixture.detectChanges();

    expect(hostComponent).toMatchSnapshot();
  });

});
