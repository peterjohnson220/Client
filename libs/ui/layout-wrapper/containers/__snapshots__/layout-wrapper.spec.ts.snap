// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Wrapper should transclude 1`] = `
<ng-component>
  
      <pf-layout-wrapper>
    <div
      class="wrapper"
    >
      
    
      <header>
        
      
        <div
          class="flex-row-nowrap"
        >
          
          
          <pf-layout-wrapper-header>
            
          
          </pf-layout-wrapper-header>
          
      
        </div>
        
    
      </header>
      
    
      <main
        class="flex-row-nowrap flex-1"
      >
        
      
        <div
          class="left-sidebar"
        >
          
        
          <pf-layout-wrapper-left-sidebar />
          
      
        </div>
        
      
        <div
          class="flex-1"
        >
          
        
          
          I am the main content
          
      
        </div>
        
      
        <div
          class="right-sidebar"
        >
          
        
          <pf-layout-wrapper-right-sidebar>
            
          
            
            I am the right sidebar content
            
        
          </pf-layout-wrapper-right-sidebar>
          
      
        </div>
        
    
      </main>
      
    
      <footer>
        
      
        <div
          class="flex-row-nowrap"
        >
          
        
          <div
            class="flex-1"
          >
            
          
            <pf-layout-wrapper-footer />
            
        
          </div>
          
      
        </div>
        
    
      </footer>
      
  
    </div>
    
  
  </pf-layout-wrapper>\`
    
</ng-component>
`;
